src/apps/profiles/models.py

```diff
@@ -34,3 +58,152 @@ def get_short_name(self):
 
     def get_full_name(self):
         return self.name
+
+    @property
+    def organized_competitions_count(self):
+        return self.organized_competitions.count() if self.organized_competitions else 0
+
+    @property
+    def datasets_count(self):
+        return self.created_datasets.count() if self.created_datasets else 0
+
+    @property
+    def tasks_count(self):
+        return self.created_tasks.count() if self.created_tasks else 0
+
+    @property
+    def solutions_count(self):
+        return self.created_solutions.count() if self.created_solutions else 0
+
+
+class Profile(models.Model):
+    """
+    Keeps track of remote users from producers
+    """
+
+    remote_id = models.IntegerField()
+    producer = models.ForeignKey(Producer, related_name='profiles', on_delete=models.CASCADE)
+    email = models.EmailField()
+    username = models.CharField(max_length=150, default=uuid.uuid4)  # Required, but not unique
+
+    user = models.ForeignKey(User, related_name='profiles', on_delete=models.SET_NULL, null=True, blank=True)
+
+    details = JSONField(null=True, blank=True)
+
+    class Meta:
+        unique_together = ['remote_id', 'producer']
+
+    @property
```

Comment:
We discussed the ESL portion of this and it should be fine. These properties are also limited to the Create/Detail serializer views for profiles so that query count does not blow up when visiting the list API endpoint for this model.